client/state/ui/section/utils.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import { SECTION_SET } from 'state/action-types';
+
+/**
+ * Given an initial state and section object key to manage, returns a reducer
+ * function which manages state for the section key.
+ *
+ * @param  {*}        initialState Initial state value
+ * @param  {String}   key          Section object key
+ * @return {Function}              Section reducer
+ */
+export function createSectionReducer( initialState, key ) {
+	return createReducer( initialState, {
+		[ SECTION_SET ]: ( state, { section } ) => {
+			if ( section && section.hasOwnProperty( key ) ) {
+				return section[ key ];
+			}
+
+			return state;
```

Comment:
>obviously if this is a bool, string, or number this doesn't apply…

Yeah, that's the case here, and I'd accounted for that in the implementation. If we were to introduce further nesting of objects (which I hope we don't), I'd rather we do that as its own reducer which would be responsible for returning same state reference on no change.